### PR TITLE
Tweak: Load nested elements styles only when they are in use [ED-15011]

### PR DESCRIPTION
**Pull request by rami-elementor**

### CODE REVIEW

**Review by KingYes — APPROVED** [review]

File: modules/nested-accordion/module.php [hunk 1]
```diff
@@ -27,5 +26,21 @@ public function __construct() {
 				'nested-elements',
 			], ELEMENTOR_VERSION, true );
 		} );
+
+		add_action( 'elementor/frontend/after_register_styles', [ $this, 'register_style' ] );
```
Comment (rotemee) [inline modules/nested-accordion/module.php:30]: @rami-elementor - the before action is "/editor" while the after is "/frontend" is it intended?

File: modules/nested-accordion/module.php [hunk 2]
```diff
@@ -27,5 +26,21 @@ public function __construct() {
...
+
+	/**
+	 * Register styles.
+	 *
+	 * @return void
+	 */
+	public function register_style() {
+		wp_register_style(
+			$this->get_name(),
+			$this->get_css_assets_url( 'frontend', 'assets/css/modules/nested-accordion/' ),
```
Comment (rotemee) [inline modules/nested-accordion/module.php:41]: @rami-elementor - is it tested in dev+prod+RTL?

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 3]
```diff
@@ -44,6 +44,10 @@ public function get_keywords() {
 		return [ 'nested', 'tabs', 'accordion', 'toggle' ];
 	}
 
+	public function get_style_depends(): array {
+		return [ 'nested-accordion' ];
```
Comment (rotemee) [inline modules/nested-accordion/widgets/nested-accordion.php:48]: @rami-elementor - is it ok that it's going to be loaded in the footer? (layout shift etc..)

File: modules/nested-tabs/module.php [hunk 4]
```diff
@@ -26,5 +26,21 @@ public function __construct() {
 				'nested-elements',
 			], ELEMENTOR_VERSION, true );
 		} );
+
+		add_action( 'elementor/frontend/after_register_styles', [ $this, 'register_style' ] );
```
Comment (rotemee) [inline modules/nested-tabs/module.php:30]: Same as above.

File: modules/nested-tabs/module.php [hunk 5]
```diff
@@ -26,5 +26,21 @@ public function __construct() {
...
+	/**
+	 * Register styles.
+	 *
+	 * @return void
+	 */
+	public function register_style() {
+		wp_register_style(
+			$this->get_name(),
+			$this->get_css_assets_url( 'frontend', 'assets/css/modules/nested-tabs/' ),
+			[],
```
Comment (rotemee) [inline modules/nested-tabs/module.php:42]: @rami-elementor - Are you sure that no-dependency is needed? I see that there is a base-tabs.scss file for example.

**Review by rotemee — CHANGES_REQUESTED** [review]

File: .grunt-config/sass.js [hunk 6]
```diff
@@ -70,6 +70,20 @@ const sass = {
 				dest: 'assets/css/modules/apps',
 				ext: '.css',
 			},
+			{
```
[the remark this answers is not in the document]
Comment (louiswol94) [reply .grunt-config/sass.js:73]: Thanks @rotemee!

I actually prefer not to add a specific new entry for every module.

If we can create a rule like `modules/*/assets/sccs/*.sccs` - or something like this, and then just place the SCSS files in the correct location in each module, that would be great.

File: modules/nested-accordion/module.php (hunk 1 above)
Comment (rami-elementor) [reply modules/nested-accordion/module.php:30]: The previous action is for JS, this one is for CSS.

When registering styles, there are specificity issues, therefor we need to position them after the all frontend styles are loaded.

Furthermore, when it was in the "_modules.scss" file, it was loaded in the end, after all fronend atyles are loaded.

File: modules/nested-accordion/widgets/nested-accordion.php (hunk 3 above)
Comment (hein-obox) [reply modules/nested-accordion/widgets/nested-accordion.php:48]:
```suggestion
		return [ $this->get_name() ];
```

File: modules/nested-accordion/module.php (hunk 1 above)
Comment (hein-obox) [reply modules/nested-accordion/module.php:30]: @rami-elementor The use of this action hook is causing some specificity problems. That is why some of the nested tabs tests are failing.

I would prefer if we can enqueue the style in the `head `of the page, but after all other styles. I am not sure if we can do that?

Comment (hein-obox) [reply modules/nested-accordion/module.php:30]: @rami-elementor @rotemee @louiswol94 
I have worked on a possible solution to enqueue the widgets in this order:
- frontend.css
- conditionally loaded widgets
- inline widget styling

Comment (rami-elementor) [reply modules/nested-accordion/module.php:30]: @hein-obox how do you suggest to load the style in the head?

Comment (hein-obox) [reply modules/nested-accordion/module.php:30]: @rami-elementor Sorry, I think I had forgotten to share my test PR: 
https://github.com/elementor/elementor/pull/27944. 

My idea was to include the active widget CSS with this hook: `'elementor/frontend/enqueue_widgets'`

**Pull request closed without merging**